jdbc/src/main/java/com/qubole/quark/jdbc/schema/dbschema/DbSchemaFactory.java

```diff
@@ -0,0 +1,154 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Sets up and organizes the planner with tables from all federated data sources.
+ * The info argument encodes all the required information (described below) as
+ * JSON.
+ * Consider an example where there are 3 data sources:
+ * <ul>
+ * <li>CANONICAL</li>
+ * <li>CUBES</li>
+ * <li>VIEWS</li>
+ * </ul>
+ * CANONICAL is the default. Each of the data sources have many schemas and
+ * tables. This class and its dependents set up such that the tables can be
+ * referred as follows:
+ * <ul>
+ * <li><i>canonical.default.lineitem</i> refers to a table in DEFAULT in CANONICAL.</li>
+ * <li><i>canonical.tpch.customers</i> refers to a table in TPCH in CANONICAL.</li>
+ * <li><i>lineitem</i> refers to a table in DEFAULT in CANONICAL.</li>
+ * <li><i>CUBES.PUBLIC.WEB_RETURNS</i> refers to a table in PUBLIC in CUBES.</li>
+ * </ul>
+ *
+ */
+public class DbSchemaFactory implements QuarkFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(DbSchemaFactory.class);
+
+  /**
+   * Creates list of QuarkSchema
+   *
+   * @param info A JSON string which represents a RootSchema and its dependent objects.
+   * @return
+   * @throws QuarkException
+   */
+  public QuarkFactoryResult create(Properties info) throws QuarkException {
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.registerModule(new GuavaModule());
+      DbCredentials dbCredentials = objectMapper
+          .readValue((String) info.getProperty("dbCredentials"), DbCredentials.class);
+
+      DBI dbi = new DBI(
+          dbCredentials.url,
+          dbCredentials.username,
+          dbCredentials.password);
+
+      Flyway flyway = new Flyway();
+      flyway.setDataSource(
+          dbCredentials.url,
+          dbCredentials.username,
+          dbCredentials.password);
+      flyway.migrate();
+
+      DSSetDAO dsSetDAO = dbi.onDemand(DSSetDAO.class);
+      JdbcSourceDAO jdbcSourceDAO = dbi.onDemand(JdbcSourceDAO.class);
+      QuboleDbSourceDAO quboleDbSourceDAO = dbi.onDemand(QuboleDbSourceDAO.class);
+      CubeDAO cubeDAO = dbi.onDemand(CubeDAO.class);
+      ViewDAO viewDAO = dbi.onDemand(ViewDAO.class);
+      MeasureDAO measureDAO = dbi.onDemand(MeasureDAO.class);
+      DimensionDAO dimensionDAO = dbi.onDemand(DimensionDAO.class);
+
+      List<DSSet> dsSets = dsSetDAO.findAll();
+      long dsSetId = dsSets.get(0).getId();
+      long defaultDataSourceId = dsSets.get(0).getDefaultDatasourceId();
+      List<JdbcSource> jdbcSources = jdbcSourceDAO.findByDSSetId(dsSetId);
+      List<QuboleDbSource> quboleDbSources = quboleDbSourceDAO.findByDSSetId(dsSetId);
+
+      ImmutableList.Builder<com.qubole.quark.planner.DataSourceSchema> schemaList =
+          new ImmutableList.Builder<>();
+
+      for (DataSource dataSource : jdbcSources) {
+        schemaList.add(new DataSourceSchema(dataSource.getProperties(defaultDataSourceId)));
```

Comment:
I understand why you are passing `defaultDataSourceId`. Not really your fault. It stinks though.